Parser/spark.py

```diff
@@ -312,12 +313,15 @@ def parse(self, tokens):
             self.states = { 0: self.makeState0() }
             self.makeState(0, self._BOF)
 
-        for i in xrange(len(tokens)):
+        i = 0
+        while i < len(tokens):
```

Comment:
You could use `enummerate()`.

Reply:
Fixed in 34ea1cf9f7be27c12156d6c9e6c19984068e44a9.